src/main/java/com/thealgorithms/sorts/WaveSort.java

```diff
@@ -0,0 +1,24 @@
+package com.thealgorithms.sorts;
+
+/**
+ * The WaveSort algorithm sorts an array so that every alternate element is greater than its adjacent elements.
+ */
+public class WaveSort implements SortAlgorithm {
+
+    @Override
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        int n = array.length;
+
+        for (int i = 0; i < n; i += 2) {
+            if (i > 0 && SortUtils.less(array[i], array[i - 1])) {
+                SortUtils.swap(array, i, i - 1);
+            }
+
+            if (i < n - 1 && SortUtils.less(array[i], array[i + 1])) {
+                SortUtils.swap(array, i, i + 1);
+            }
+        }
+
+        return array;
+    }
+}
```

Comment:
I would not derive the `WaveSort` from `SortAlgorithm`. Each class implementing `SortAlgorithm` should be tested by a class derived from `SortingAlgorithmTest` - in this case it will simply not work. Furthermore `WaveSort` is so much different that the usual _sorting algorithm_ (e.g. the output is not unique).

Reply:
I agree, this is a very specialized sort. The idea was to add it as an addition to the existing WiggleSort.

This class already contains triColorSort and wiggleSort methods.

We can either add it as a separate method within this class or decline the pull request. 

What to do with this?

Reply:
Good point.

Since `WiggleSort` implements `SortAlgorithm`, then `WaveSort` also should. However, I would suggest that `WaveSort` additionally should have a `public` method `isWaveSorted`.

We can discuss in the future what `SortAlgorithm` should really be.

Reply:
This needs a bit clarification. Currently, the sort package contains some specialized sorts. Additionally, the package includes TopologicalSort, which logically belongs in the graph section. Also these specialized sorts, such as WiggleSort and now WaveSort, should be categorized under a different section for other sorts.